### PR TITLE
*: use new MyDecimal implementation.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @tiancaiamao PTAL

**Comment by shenli** [comment]
@coocood Have you passed tidb-test?

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
@shenli yes.

File: executor/aggregate_xapi_test.go [hunk 1]
```diff
@@ -337,22 +337,22 @@ func (s *testAggFuncSuite) TestXAPIAvg(c *C) {
...
 	// Third row: 16.5000
 	row, err = agg.Next()
 	c.Assert(err, IsNil)
 	c.Assert(row, NotNil)
 	val, err = evaluator.Eval(ctx, fc)
 	c.Assert(err, IsNil)
-	d := mysql.NewDecimalFromFloat(float64(16.5))
-	d.SetFracDigits(4) // For div operator, default frac is 4.
+	d := mysql.NewDecFromFloatForTest(16.5)
+	c.Logf("d1 %s, d2 %s, cmp %d", d.String(), val.GetMysqlDecimal().String(), val.GetMysqlDecimal().Compare(d))
```
Comment (tiancaiamao) [inline executor/aggregate_xapi_test.go]: clean up debug log?

File: tablecodec/tablecodec.go [hunk 2]
```diff
@@ -317,23 +314,6 @@ func Unflatten(datum types.Datum, ft *types.FieldType, inIndex bool) (types.Datu
 		dur := mysql.Duration{Duration: time.Duration(datum.GetInt64())}
 		datum.SetValue(dur)
 		return datum, nil
-	case mysql.TypeNewDecimal:
```
Comment (tiancaiamao) [inline tablecodec/tablecodec.go]: both the old and new decimal gone?

Comment (tiancaiamao) [inline tablecodec/tablecodec.go]: oh, it's flatten and Unflattenignore me...

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**